src/main/java/fr/inria/coming/codefeatures/RepairPatternFeatureAnalyzer.java

```diff
@@ -0,0 +1,326 @@
+	 static int insertIfFalse = 0;  //
+
+	 static int ifTrue = 0;  //if true to replace if condition
+	 static int condLogicReduce = 0;  //if false to replace if condition
+	 static int insertBooleanLiteral = 0; //a varaible is replaced by true or false
+	 static int insertNewConstLiteral = 0; // const, false, true
+	 static int UpdateLiteral = 0 ;
+	 static int removeNullinCond = 0 ; //remove null in if
+	 static int	patchedFileNo=0;
+	 static int	rmLineNo=0;
+	 static int	addLineNo=0;
+	 static int	 addThis=0;
+
+
+
+	public static JsonObject analyze(String folderPath, Diff diff, String targetFile) {
+		
+		dupArgsInvocation = 0;
+		updIfFalse = 0; 
+		insertIfFalse = 0;
+		ifTrue = 0;  
+		condLogicReduce = 0;  
+		insertBooleanLiteral = 0; 
+		insertNewConstLiteral = 0;
+		UpdateLiteral = 0;
+		removeNullinCond = 0;
+		patchedFileNo=0;
+		rmLineNo=0;
+		addLineNo=0;
+		addThis=0;
+		
+		Map<String, File> filePaths = new P4JFeatureAnalyzer().processFilesPair(new File(folderPath),targetFile);
+		File src = filePaths.get("src");
+		File target = filePaths.get("target");
+		
+		
+      patchAnalysis(folderPath);
+	  // if(false) in one line conditional replacement
+	  ifFalseAnalyzer(folderPath, diff, target);	
+	  ifTrueAnalyzer(folderPath, diff, target);	
+	  //dupArgsInvocation if duplicated arguments in one method invocation 
+	  dupArgsInvocation(folderPath, diff, target);
+	  condLogicReduceAnalyzer(folderPath, diff, src,target);
+	  insertLiteralAnalyzer(folderPath, diff,src, target);
+	  rmNullAnalyzer(folderPath, diff, src,target);
+	  addThisAnalyzer(folderPath, diff, src,target);
+
+
+      JsonObject jsonObject = new JsonObject();
+      jsonObject.addProperty("updIfFalse", updIfFalse);
+      jsonObject.addProperty("insertIfFalse", insertIfFalse);
+      jsonObject.addProperty("ifTrue", ifTrue);
+      jsonObject.addProperty("dupArgsInvocation", dupArgsInvocation);
+      jsonObject.addProperty("condLogicReduce", condLogicReduce);
+      jsonObject.addProperty("insertNewConstLiteral", insertNewConstLiteral);
+      jsonObject.addProperty("insertBooleanLiteral", insertBooleanLiteral);
+      jsonObject.addProperty("UpdateLiteral", UpdateLiteral);
+      jsonObject.addProperty("removeNullinCond", removeNullinCond);
+      jsonObject.addProperty("patchedFileNo", patchedFileNo);
+      jsonObject.addProperty("rmLineNo", rmLineNo);
+      jsonObject.addProperty("addLineNo", addLineNo);
+      jsonObject.addProperty("addThis", addThis);
+
+	  return jsonObject;
+
+	}
+
+	private static void addThisAnalyzer(String folderPath, Diff diff, File src, File target) {
+		List<Operation> operations = diff.getRootOperations();
+		for (Operation opt : operations) {
+			if (opt.toString().contains("Insert") ) {
+				String newvar = opt.toString().split(":")[1].split("\n\t")[1];
+				if(newvar.contains("this")) {
+					addThis = 1;
+				}
+			}
+		}
+		
+	}
+
+	private static void patchAnalysis(String folderPath) {
```

Comment:
It seems that this method only computes the number of lines added/removed/changed. 
Can we put a more representative name?